students/woogyeong/westagram/users/views.py

```diff
@@ -1 +1,54 @@
-from django.shortcuts import render
+# python built-in module
+import json
+import re
+
+# 외부 module
+from django.http  import JsonResponse
+from django.views import View
+
+# 사용자 module (직접 작성한 module)
+from users.models import User
+
+class SignUpView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        
+        def validate_email(email=data['email']):
+            email_regex      = re.compile(r"^[a-zA-Z]+[!#$%&'*+-/=?^_`(){|}~]*[a-zA-Z0-9]*@[\w]+\.[a-zA-Z0-9-]+[.]*[a-zA-Z0-9]+$")
+            email_validation = email_regex.match(email)
+            
+            if email_validation:
+                return True
+            else:
+                return False
+        
+        def validate_password(password=data['password']):
+            password_regex = re.compile(r'([\w!@#$%^&*(),.?\":{}|<>]+){8}')
+            password_valid = password_regex.match(password)
+            
+            if password_valid:
+                return True
+            else:
+                return False
+        
+        try:
+            if User.objects.filter(email=data['email']).exists():
+                return JsonResponse({'message': 'USER_EXISTS'}, status=409)
+            else:
+                if validate_email() == True and validate_password() == True:
+                    pass
+                elif validate_email() == False:
+                    return JsonResponse({'message': 'INVALID_EMAIL'}, status=409)
+                elif validate_password() == False:
+                    return JsonResponse({'message': 'INVALID_PASSWORD'}, status=409)
```

Comment:
```suggestion
      if not re.match('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$', email)
                return JsonResponse({'message':'Email format is not valid'}, status=400)

        if not re.match('^(?=.*[A-Za-z])(?=.*\d)(?=.*[?!@#$%*&])[A-Za-z\d?!@#$%*&]{8,}$', password):
            return JsonResponse({'message':'Password format is not valid'}, status=400)
        
        if User.objects.filter(email=data['email']).exists():
                return JsonResponse({'message': 'USER_EXISTS'}, status=400)
```
1. match를 사용할 경우 compile 없이 바로 사용 가능합니다.
2. 정규 표현식 같은 경우 리뷰에는 직접 바로 넣어지만, 상수화 해도 좋을 것 같습니다.
3. validation error의 경우 보통 400을 많이 사용합니다. 409는 일반적으로 put method를 사용할 때 타겟 데이터랑 수정하려는 정보랑 충돌이 일어 났을 때 사용합니다.

Reply:
경훈님 제가 함수를 만들어서 써본 경험이 거의 없어서 validators 라는 모듈을 만들어서 해당 함수들을 import 해봤습니다.
그리고 두번째 말씀하신 부분 상수화하는 것을 validators 모듈에 했는데 제가 이해한게 맞을까요~??

Reply:
경훈님 제가 3번 미션하기 전에 branch를 따지 않아서 이 PR과 commit 삭제하고 새로 PR을 해도 될까요?